### PR TITLE
Don't allow to be run from outside Applications folder

### PR DESCRIPTION
**Pull request by dpgraham**
* Don't allow to be run from outside Applications folder 
    * Added 'WrongFolder' component that is shown if user is not in applications folder on Mac that shows a 'Move to' button that moves the application programatically
    * When user clicks 'Move to' button, send message to main thread which calls `app.moveToApplicationsFolder`
    * Skip the wrong folder warning if it's an E2E test or if it's in dev
* Upgraded to Electron 1.8.1 (note that 1.8 is in beta so we'll have to hold off on cherry-picking this to a release until it's out of beta)
* Replaced `electron-prebuilt` with `electron`
* Fixed broken NPM scripts that resulted because `babel-register` is no longer in `.bin`

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
finally we got some time to have this feature 👍

File: app/renderer/index.js [hunk 1]
```diff
@@ -3,17 +3,37 @@ import { render } from 'react-dom';
...
 import configureStore from './store/configureStore';
 import './styles/app.global.css';
 import './styles/github-gist-theme.global.css';
 
+const {app} = electron.remote;
+const isDev = process.env.NODE_ENV === 'development';
 const store = configureStore();
 const history = syncHistoryWithStore(hashHistory, store);
 
+function shouldShowWrongFolderComponent () {
```
Comment (mykola-mokhnach) [inline app/renderer/index.js]: should we consider operating systems different from Mac OS?

File: app/renderer/components/WrongFolder/WrongFolder.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export default class WrongFolder extends Component {
+  // This tells the main thread to move this to applications folder which will cause the app to close and restart
+  moveToApplicationsFolder () {
+    ipcRenderer.send('appium-move-to-applications-folder');
+  }
+
+  render () {
+    return <div className={WrongFolderStyles['wrong-folder']}>
+      <div>
+        <div>Appium Desktop should be run from the Applications folder</div>
```
Comment (mykola-mokhnach) [inline app/renderer/components/WrongFolder/WrongFolder.js]: Do we show the same message on Linux/Windows?

File: app/renderer/index.js (hunk 1 above)
Comment (dpgraham) [reply app/renderer/index.js]: Yes. My first commit didn't account for that and threw a 'not a function error' on the Windows build (via AppVeyor). So I added the check that the function exists which accounts for non Mac.

File: app/renderer/components/WrongFolder/WrongFolder.js (hunk 2 above)
Comment (dpgraham) [reply app/renderer/components/WrongFolder/WrongFolder.js]: No the function only applies to Mac.

File: app/renderer/index.js (hunk 1 above)
Comment (mykola-mokhnach) [reply app/renderer/index.js]: > So I added the check that the function exists which accounts for non Mac.

Can you please show which line of this PR does this OS verification?

Comment (dpgraham) [reply app/renderer/index.js]: It doesn't verify the OS explicitly. It verifies that `isInApplicationsFolder` is defined. It's not defined in non Mac operating systems which I discovered through failed AppVeyor test from previous commits.

Comment (mykola-mokhnach) [reply app/renderer/index.js]: then it worth adding a comment

File: app/renderer/components/WrongFolder/WrongFolder.js (hunk 2 above)
Comment (mykola-mokhnach) [reply app/renderer/components/WrongFolder/WrongFolder.js]: do we exit the app if this dialog is canceled?

Comment (dpgraham) [reply app/renderer/components/WrongFolder/WrongFolder.js]: Yeah. Do you think we should give them the option to proceed without moving the folder?

Comment (mykola-mokhnach) [reply app/renderer/components/WrongFolder/WrongFolder.js]: definitely not ;)

**Pull request merged**